torch/_dynamo/output_graph.py

```diff
@@ -2155,16 +2158,27 @@ def create_graph_input(
             #     can be created while tracing. So we use track_unbacked_symbols will intercept
             #     at wrap_fx_proxy, and try to bind the unbacked symbols immediately after they're
             #     created.
-            #  3. subgraph will also lifted basic symbols in compound exprs.
+            #  3. subgraph will also lifted basic symbols in compound exprs of tensor shape.
             #     For example, if an input to subgraph takes size [s1+s2//8], we'll look for the
             #     the free symbols in the sizes and lift as inputs similar to 1 in _lift_symbols_in_symint)
-            # Do not call this function direclty. This is used in create_graph_input for
-            # automatically lifting the free basic symbols in example_value.
+            #  4. When create_graph_input for a SymInt, if the symint is a basic symbol, we'll track it
+            #     in bound_symbols so that we don't lift the same basic symbol twice. When the symint is a
+            #     compound expr, we'll just create the proxy for the compouned expr but not lift its basic symbols.
+            # Also see NOTE: [Export inputs must be explicitly passed in]
             is_strict_export = self.is_export
             is_non_strict_export = torch.compiler.is_compiling()
-            if not is_strict_export and not is_non_strict_export:
-                if isinstance(example_value, torch.Tensor):
-                    self._lift_basic_symbols(example_value, source)
+            if (
+                not is_strict_export
+                and not is_non_strict_export
+                and isinstance(example_value, torch.Tensor)
+            ):
+                self._lift_basic_symbols(example_value, source)
+
+            # Bound the symbol to ph if example_value is a SymInt with basic symbol.
```

Comment:
Simplified the bound_symbols maitaining logic for create_graph_input path: we just bound basic symbols whenever we create a basic symbol placeholder.